src/pbt/v2/project/components/airflow_jobs.py

```diff
@@ -0,0 +1,443 @@
+import os
+import zipfile
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from typing import Dict, Optional, List
+
+import yaml
+
+from src.pbt.v2.client.airflow_client.airflow_utility import create_airflow_client
+from src.pbt.v2.project.components.databricks_jobs import FABRIC_UID
+from src.pbt.v2.project.project_parser import ProjectParser
+from src.pbt.v2.project_models import StepMetadata
+from src.pbt.v2.project_config import JobInfo, ProjectConfig
+from src.pbt.v2.utility import Either, generate_secure_content, calculate_checksum
+
+
+def does_dag_file_exist(rdc: Dict[str, str]) -> bool:
+    return rdc is not None and 'dag.py' in rdc
+
+
+def filter_job_files(rdc: Dict[str, str]):
+    filtered_files = {file_name: file_content for file_name, file_content in rdc.items()
+                      if
+                      file_name == 'dag.py' or '__init__.py' or file_name == 'prophecy-job.json' or
+                      'tasks/' in file_name}
+
+    return dict(sorted(filtered_files.items()))
+
+
+def sanitize_job(job_id) -> str:
+    return ('_' if job_id is None else job_id).split('/')[-1].replace('\\W', '_')
+
+
+def get_zipped_dag_name(dag_name: str):
+    return f"/tmp/{dag_name}.zip"
+
+
+def zip_folder(folder_path, output_path):
+    with zipfile.ZipFile(output_path, 'w', zipfile.ZIP_DEFLATED) as zipf:
+        for root, dirs, files in os.walk(folder_path):
+            for file in files:
+                abs_file_path = os.path.join(root, file)
+                zipf.write(abs_file_path, abs_file_path[len(folder_path):])
+
+
+class AirflowJobDataProcessor:
+
+    def __init__(self, job_pbt: dict, prophecy_job_yaml: str, rdc: Dict[str, str], sha: Optional[str]):
+        self.job_pbt = job_pbt
+        self.prophecy_job_yaml = prophecy_job_yaml
+        self.rdc = rdc
+        self.sha = sha
+        self.prophecy_job_json_dict = {}
+
+        self._initialize_prophecy_job_json()
```

Comment:
can we inline this method here, so that all instance-level fields are initialised in one place?

Reply:
wil do return.